pkgs/http/lib/src/io_client.dart

```diff
@@ -53,17 +56,33 @@ class IOClient extends BaseClient {
 
     var stream = request.finalize();
 
+    ByteStream? handledStream;
+
+    var contentLength = request.contentLength;
+    if (request.onUploadProgress != null && contentLength != null) {
+      var load = 0;
+      handledStream =
+          ByteStream(stream.transform(StreamTransformer.fromBind((d) async* {
+        await for (var data in d) {
+          load += data.length;
+          request.onUploadProgress!.call(contentLength, load);
```

Comment:
No need for `.call` here, just call the function directly.

Reply:
Done